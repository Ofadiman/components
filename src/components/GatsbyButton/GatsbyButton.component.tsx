import React from 'react'

import { StyledGatsbyButton } from './GatsbyButton.styles'
import { GatsbyButtonProps } from './GatsbyButton.types'

export const GatsbyButton: FCC<GatsbyButtonProps> = ({ onClick, className, isSmall, children }) => (
  <StyledGatsbyButton className={className} isSmall={isSmall} onClick={onClick}>
    {children}
  </StyledGatsbyButton>
)
