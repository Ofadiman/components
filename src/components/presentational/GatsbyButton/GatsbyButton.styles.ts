import { makeStyles } from '@material-ui/core/styles'

export const useGatsbyButtonStyles = makeStyles(
  () => ({
    '@keyframes stripe-animation-medium': {
      from: {
        backgroundPosition: '0 0'
      },
      to: {
        backgroundPosition: '64px 128px'
      }
    },
    '@keyframes stripe-animation-small': {
      from: {
        backgroundPosition: '0 0'
      },
      to: {
        backgroundPosition: '32px 64px'
      }
    },
    medium: {
      '&:focus': {
        boxShadow: '0 0 0 3px hsl(270, 100%, 80%)'
      },
      '&:hover': {
        animation: '$stripe-animation-medium 2.8s linear infinite',
        backgroundSize: '64px 64px'
      },
      borderRadius: 18,
      fontSize: '3rem',
      height: 60,
      width: 300
    },
    root: {
      backgroundColor: 'hsl(270, 100%, 60%)',
      backgroundImage: `linear-gradient(
          135deg,
          rgba(0, 0, 0, 0.1) 25%,
          transparent 25%,
          transparent 50%,
          rgba(0, 0, 0, 0.1) 50%,
          rgba(0, 0, 0, 0.1) 75%,
          transparent 75%,
          transparent
        );`,
      backgroundSize: '0 0',
      border: 'none',
      color: 'white',
      cursor: 'pointer',
      outline: 'none'
    },
    small: {
      '&:focus': {
        boxShadow: '0 0 0 2px hsl(270, 100%, 80%)'
      },
      '&:hover': {
        animation: '$stripe-animation-small 2.8s linear infinite',
        backgroundSize: '32px 32px'
      },
      borderRadius: 12,
      fontSize: '2rem',
      height: 40,
      width: 200
    }
  }),
  {
    name: 'GatsbyButton'
  }
)
