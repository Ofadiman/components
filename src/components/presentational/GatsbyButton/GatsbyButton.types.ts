export interface GatsbyButtonProps {
  className?: string
  onClick: () => void
  size?: 'small' | 'medium'
  title: string
}
