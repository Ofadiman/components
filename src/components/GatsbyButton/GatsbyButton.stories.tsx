import { storiesOf } from '@storybook/react'
import React from 'react'

import { storybookLog } from '../../utils/functions/storybookLog/storybookLog.function'
import { GatsbyButton } from './GatsbyButton.component'

storiesOf('GatsbyButton', module)
  .add('Default', () => <GatsbyButton onClick={storybookLog('Regular button click')}>{'Regular button'}</GatsbyButton>)
  .add('Small', () => (
    <GatsbyButton isSmall={true} onClick={storybookLog('Small button click')}>
      {'Small button'}
    </GatsbyButton>
  ))
