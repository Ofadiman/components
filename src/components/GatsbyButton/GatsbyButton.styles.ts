import styled, { css, keyframes } from 'styled-components'

import { StyledGatsbyButtonProps } from './GatsbyButton.types'

const stripeAnimationSmall = keyframes`
  from {
    background-position: 0 0;
  }
  to {
    background-position: 32px 64px;
  }
`

const stripeAnimationRegular = keyframes`
  from {
    background-position: 0 0;
  }
  to {
    background-position: 64px 128px;
  }
`

export const StyledGatsbyButton = styled.button<StyledGatsbyButtonProps>`
  ${({ theme, isSmall }) => css`
    background-color: ${theme.palette.primary.main};
    background-size: ${isSmall ? '32px 32px' : '64px 64px'};
    border: none;
    border-radius: ${theme.spacing(4)};
    box-shadow: ${theme.shadowing(2)};
    color: ${theme.palette.primary.contrastText};
    cursor: pointer;
    font-size: ${isSmall ? '1.75rem' : '2.5rem'};
    height: ${isSmall ? '40px' : '60px'};
    outline: none;
    width: ${isSmall ? '200px' : '300px'};
    &:focus,
    &:active {
      box-shadow: 0 0 0 ${isSmall ? '2px' : '3px'} ${theme.palette.secondary.light};
    }

    ${theme.mediaQueries.tablet} {
      &:hover {
        animation: ${isSmall ? stripeAnimationSmall : stripeAnimationRegular} 2.8s linear infinite;
        background-image: linear-gradient(
          135deg,
          rgba(0, 0, 0, 0.1) 25%,
          transparent 25%,
          transparent 50%,
          rgba(0, 0, 0, 0.1) 50%,
          rgba(0, 0, 0, 0.1) 75%,
          transparent 75%,
          transparent
        );
      }
    }
  `}
`
