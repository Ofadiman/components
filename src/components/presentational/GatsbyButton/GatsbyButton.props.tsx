import { storybookLog } from '../../../functions/storybookLog/storybookLog'
import { ComponentStoriesProps } from '../../../types/ComponentStoriesProps.type'
import { GatsbyButtonProps } from './GatsbyButton.types'

export const gatsbyButtonStoryProps: ComponentStoriesProps<GatsbyButtonProps> = [
  { onClick: storybookLog('small button click'), size: 'small', title: 'Small' },
  { onClick: storybookLog('medium button click'), size: 'medium', title: 'Medium' }
]
