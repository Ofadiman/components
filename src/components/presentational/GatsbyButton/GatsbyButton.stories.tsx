import { storiesOf } from '@storybook/react'
import React from 'react'

import { GatsbyButton } from './GatsbyButton'
import { gatsbyButtonStoryProps } from './GatsbyButton.props'

const clientApi = storiesOf('Components/Presentational/GatsbyButton', module)

gatsbyButtonStoryProps.forEach(({ title, ...props }) => {
  clientApi.add(title, () => <GatsbyButton {...props} />)
})
