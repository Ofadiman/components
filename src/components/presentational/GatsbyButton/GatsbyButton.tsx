import clsx from 'clsx'
import React from 'react'

import { useGatsbyButtonStyles } from './GatsbyButton.styles'
import { GatsbyButtonProps } from './GatsbyButton.types'

export const GatsbyButton: FC<GatsbyButtonProps> = ({ className, onClick, size = 'medium' }) => {
  const classes = useGatsbyButtonStyles()

  return (
    <button
      className={clsx(classes.root, className, {
        [classes.small]: size === 'small',
        [classes.medium]: size === 'medium'
      })}
      onClick={onClick}
      type={'button'}
    >
      {'Click me!'}
    </button>
  )
}
