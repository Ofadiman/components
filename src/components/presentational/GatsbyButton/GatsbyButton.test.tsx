import React from 'react'

import { renderComponent } from '../../../functions/renderComponent/renderComponent'
import { GatsbyButton } from './GatsbyButton'
import { gatsbyButtonStoryProps } from './GatsbyButton.props'

describe('GatsbyButton component', () => {
  gatsbyButtonStoryProps.forEach(({ title, ...props }) => {
    test(`match snapshot for ${title}`, () => {
      const { asFragment } = renderComponent(<GatsbyButton {...props} />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
