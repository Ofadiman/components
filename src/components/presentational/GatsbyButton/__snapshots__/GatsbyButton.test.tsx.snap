// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GatsbyButton component match snapshot for Medium 1`] = `
<DocumentFragment>
  <button
    class="GatsbyButton-root-9 GatsbyButton-medium-8"
    type="button"
  >
    Click me!
  </button>
</DocumentFragment>
`;

exports[`GatsbyButton component match snapshot for Small 1`] = `
<DocumentFragment>
  <button
    class="GatsbyButton-root-4 GatsbyButton-small-5"
    type="button"
  >
    Click me!
  </button>
</DocumentFragment>
`;
